rusoto/credential/src/profile.rs

```diff
@@ -20,6 +20,18 @@ pub struct ProfileProvider {
 }
 
 impl ProfileProvider {
+
+    /// Get the default profile name:
+    /// 1. if set use value from environment variable ```AWS_PROFILE```
+    /// 2. otherwise return ```"default"```
+    /// see https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html.
+    fn default_profile_name() -> String {
+         match env_var("AWS_PROFILE") {
```

Comment:
This looks to be a case for [`Result::unwrap_or_else`](https://doc.rust-lang.org/1.16.0/std/result/enum.Result.html#method.unwrap_or_else)!